### PR TITLE
PIN-4220 tenant-process: Added Mail Id

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4219](https://github.com/pagopa/interop-be-tenant-management/pull/67) to compile

[PIN-4219]: https://pagopa.atlassian.net/browse/PIN-4219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala [hunk 1]
```diff
@@ -135,4 +135,28 @@ final case class TenantManagementServiceImpl(
...
     readModel: ReadModelService
   ): Future[PaginatedResult[PersistentTenant]] = ReadModelTenantQueries.listConsumers(name, producerId, offset, limit)
+
+  override def addTenantMail(tenantId: UUID, mailSeed: MailSeed)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = withHeaders { (bearerToken, correlationId) =>
+    val request = tenantApi.addTenantMail(xCorrelationId = correlationId, tenantId = tenantId, mailSeed = mailSeed)(
+      BearerToken(bearerToken)
+    )
+    invoker.invoke(request, s"Adding mail ${mailSeed.address} to tenant $tenantId").recoverWith {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala]: Potential sensitive info
```suggestion
    invoker.invoke(request, s"Adding mail with id ${mailSeed.id} to tenant $tenantId").recoverWith {
```

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala [hunk 2]
```diff
@@ -886,4 +848,42 @@ final case class TenantApiServiceImpl(
       getTenantByExternalIdResponse[Tenant](operationLabel)(getTenantByExternalId200)
     }
   }
+
+  override def addTenantMail(tenantId: String, mailSeed: MailSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE, INTERNAL_ROLE) {
+    val operationLabel = s"Adding mail ${mailSeed.address} to Tenant $tenantId"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]: Potential sensitive info
```suggestion
    val operationLabel = s"Adding mail of kind ${mailSeed.kind} to Tenant $tenantId"
```

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala [hunk 3]
```diff
@@ -886,4 +848,42 @@ final case class TenantApiServiceImpl(
       getTenantByExternalIdResponse[Tenant](operationLabel)(getTenantByExternalId200)
     }
   }
+
+  override def addTenantMail(tenantId: String, mailSeed: MailSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE, INTERNAL_ROLE) {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]: @beetlecrunch are there any restrictions on roles who may add the mail?

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala [hunk 4]
```diff
@@ -886,4 +848,42 @@ final case class TenantApiServiceImpl(
...
+
+    onComplete(result) {
+      addTenantMailResponse[Unit](operationLabel)(_ => addTenantMail204)
+    }
+  }
+
+  override def deleteTenantMail(tenantId: String, mailId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE, INTERNAL_ROLE) {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]: @beetlecrunch same here on restrictions

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala (hunk 3 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]:
```suggestion
  ): Route = authorize(ADMIN_ROLE, API_ROLE) {
```

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala (hunk 4 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]:
```suggestion
  ): Route = authorize(ADMIN_ROLE, API_ROLE) {
```

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**